appgate/attrs.py

```diff
@@ -79,7 +90,57 @@ def dump_datetime(v: datetime.datetime) -> str:
     return v.isoformat(timespec="milliseconds").replace("+00:00", "Z")
 
 
-def get_dumper(platform_type: PlatformType):
+def k8s_name(name: str) -> str:
+    # This is ugly but we need to go from a bigger set of strings
+    # into a smaller one :(
+    return re.sub("[^a-z0-9-.]+", "-", name.strip().lower())
+
+
+def k8s_dumper(
+    dumper: Dumper,
+    entity: Entity_T,
+    api_spec: APISpec,
+    strict: bool = True,
+    resolution_conflicts: Dict[str, List[MissingFieldDependencies]] | None = None,
+) -> Dict[str, Any]:
+    entity_kind = entity.__class__.__qualname__
+    annotations = {}
+    # Check if entity has some resolution conflicts
+    fs = ";".join(
+        f.field_path for f in (resolution_conflicts or {}).get(entity.name, [])
+    )
+    if fs:
+        annotations[K8S_FIELD_WITH_IDS_ANNOTATION] = fs
+    if has_id(entity):
+        annotations[K8S_ID_ANNOTATION] = entity.id
+        entity = evolve(
+            entity,
+            appgate_metadata=evolve(entity.appgate_metadata, uuid=entity.id),
+        )
+    if is_singleton(entity):
+        entity_name = k8s_name(entity_kind)
+    elif has_name(entity):
+        entity_name = k8s_name(entity.name)
+    elif strict:
+        raise AppgateTypedloadException(
+            "Unable to dump entity: name/id field is missing",
+            platform_type=PlatformType.K8S,
+        )
```

Comment:
Do we need to fix the if-elif-else statement here? I think we could fall into the scenario where we `has_name = True` but `has_id = False` then it would pass this condition.

Reply:
Ah. I guess we don't fall into that scenario because if the `id` field doesn't exist, we automatically generate one for the entity.

Reply:
id is always created, indeed that stuff is mostly for unit tests where the test entities dont have always name or ids :D